File: mario.c
#include <stdio.h>
#include <cs50.h>

void print_spaces(int n);
void print_hash(int n);

int main(void)
{
    int height = 0;
    int row_number = 1;
    bool is_valid = false;

    // prompt and validate user input
    do
    {
        printf("Height: ");
        height = get_int();
        if (height >= 0 && height <= 23)
        {
            is_valid = true;
        }
        else
        {
            is_valid = false;
        }
    }
    while (!is_valid);

    // draw the half pyramid

// checking base condition
    if (height == 0)
    {
        printf("");
    }
    else
    {
        while   (row_number <= height)
        {
            // calling print_spaces and print_hash depending on row
            print_spaces(height - row_number);
            print_hash(row_number + 1);
            printf("\n");
            row_number++;
        }
    }


}

// prints n number of spaces
void print_spaces(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf(" ");
    }
}

// prints n number of hashes
void print_hash(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("#");
    }
}

File: mario.py
from cs50 import get_int

# checking user input for non negative height

while True:
    height = get_int("Height: ")
    if height >= 0 and height <= 23:
        break

# initializing number of space and brick in pyramid

column = height + 1
space = column - 2
brick = column - space
row = 1

# printing pyrmaid for a given height

if height > 0:
    while True:
        print(" " * space, end="")
        print("#" * brick)
        row += 1
        space -= 1
        brick += 1
        if row > height:
            break
else:
    print(end="")